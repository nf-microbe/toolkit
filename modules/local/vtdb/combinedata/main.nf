process VTDB_COMBINEDATA {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta)     , path(fasta)
    tuple val(meta2)    , path(trfinder_stats)
    tuple val(meta3)    , path(genomad_scores)
    tuple val(meta4)    , path(genomad_genes)
    tuple val(meta5)    , path(genomad_taxa)
    tuple val(meta6)    , path(busco_hmms)
    tuple val(meta7)    , path(plasmid_hmms)
    tuple val(meta8)    , path(virus_hmms)
    tuple val(meta7)    , path(completeness)
    tuple val(meta8)    , path(contamination)
    tuple val(meta9)    , path(tantan)
    tuple val(meta10)   , path(nuc_stats)

    output:
    tuple val(meta), path("${prefix}_combined_data.tsv")    , emit: tsv
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args                    = task.ext.args ?: ''
    def trfinder_input          = trfinder_stats ? "--trfinder ${trfinder_stats}" : ""
    def genomad_scores_input    = genomad_scores ? "--genomad_scores ${genomad_scores}" : ""
    def genomad_genes_input     = genomad_genes ? "--genomad_genes ${genomad_genes}" : ""
    def genomad_taxa_input      = genomad_taxa ? "--genomad_taxa ${genomad_taxa}" : ""
    def busco_hmms_input        = busco_hmms ? "--busco_hmms ${busco_hmms}" : ""
    def plasmid_hmms_input      = plasmid_hmms ? "--plasmid_hmms ${plasmid_hmms}" : ""
    def virus_hmms_input        = virus_hmms ? "--virus_hmms ${virus_hmms}" : ""
    def completeness_input      = completeness ? "--completeness ${completeness}" : ""
    def contamination_input     = contamination ? "--contamination ${contamination}" : ""
    def tantan_input            = tantan ? "--tantan ${tantan}" : ""
    def nucleotide_stats_input  = nuc_stats ? "--sequence_stats ${nuc_stats}" : ""
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    combinedata.py \\
        --input ${fasta} \\
        ${trfinder_input} \\
        ${genomad_scores_input} \\
        ${genomad_genes_input} \\
        ${genomad_taxa_input} \\
        ${busco_hmms_input} \\
        ${plasmid_hmms_input} \\
        ${virus_hmms_input} \\
        ${completeness_input} \\
        ${contamination_input} \\
        ${tantan_input} \\
        ${nucleotide_stats_input} \\
        --output ${prefix}_combined_data.tsv \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_combined_data.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """
}
