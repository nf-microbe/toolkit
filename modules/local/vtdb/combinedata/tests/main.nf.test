nextflow_process {

    name "Test Process: COMBINEDATA"
    script "../main.nf"
    process "COMBINEDATA"

    test("all files") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.tsv", checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_aggregated_classification.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_genes.tsv", checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_taxonomy.tsv", checkIfExists: true)
                ]
                input[5] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_extra_hmm_hits.tsv", checkIfExists: true)
                ]
                input[6] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_completeness.tsv", checkIfExists: true)
                ]
                input[7] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_contamination.tsv", checkIfExists: true)
                ]
                input[8] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_tantan.tsv", checkIfExists: true)
                ]
                input[9] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_nuc_stats.tsv", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("no trfinder") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.fasta.gz", checkIfExists: true)
                ]
                input[1] = Channel.of([ [:], [] ])
                input[2] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_aggregated_classification.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_genes.tsv", checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_taxonomy.tsv", checkIfExists: true)
                ]
                input[5] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_extra_hmm_hits.tsv", checkIfExists: true)
                ]
                input[6] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_completeness.tsv", checkIfExists: true)
                ]
                input[7] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_contamination.tsv", checkIfExists: true)
                ]
                input[8] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_tantan.tsv", checkIfExists: true)
                ]
                input[9] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_nuc_stats.tsv", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("no genomad") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.tsv", checkIfExists: true)
                ]
                input[2] = Channel.of([ [:], [] ])
                input[3] = Channel.of([ [:], [] ])
                input[4] = Channel.of([ [:], [] ])
                input[5] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_extra_hmm_hits.tsv", checkIfExists: true)
                ]
                input[6] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_completeness.tsv", checkIfExists: true)
                ]
                input[7] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_contamination.tsv", checkIfExists: true)
                ]
                input[8] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_tantan.tsv", checkIfExists: true)
                ]
                input[9] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_nuc_stats.tsv", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("no pyhmmer") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.tsv", checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_aggregated_classification.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_genes.tsv", checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_taxonomy.tsv", checkIfExists: true)
                ]
                input[5] = Channel.of([ [:], [] ])
                input[6] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_completeness.tsv", checkIfExists: true)
                ]
                input[7] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_contamination.tsv", checkIfExists: true)
                ]
                input[8] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_tantan.tsv", checkIfExists: true)
                ]
                input[9] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_nuc_stats.tsv", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("no checkv") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.tsv", checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_aggregated_classification.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_genes.tsv", checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_taxonomy.tsv", checkIfExists: true)
                ]
                input[5] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_extra_hmm_hits.tsv", checkIfExists: true)
                ]
                input[6] = Channel.of([ [:], [] ])
                input[7] = Channel.of([ [:], [] ])
                input[8] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_tantan.tsv", checkIfExists: true)
                ]
                input[9] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_nuc_stats.tsv", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("no nucleotide stats") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.tsv", checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_aggregated_classification.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_genes.tsv", checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_taxonomy.tsv", checkIfExists: true)
                ]
                input[5] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_extra_hmm_hits.tsv", checkIfExists: true)
                ]
                input[6] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_completeness.tsv", checkIfExists: true)
                ]
                input[7] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_contamination.tsv", checkIfExists: true)
                ]
                input[8] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_tantan.tsv", checkIfExists: true)
                ]
                input[9] = Channel.of([ [:], [] ])
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("no tantan") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.tsv", checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_aggregated_classification.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_genes.tsv", checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_taxonomy.tsv", checkIfExists: true)
                ]
                input[5] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_extra_hmm_hits.tsv", checkIfExists: true)
                ]
                input[6] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_completeness.tsv", checkIfExists: true)
                ]
                input[7] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_contamination.tsv", checkIfExists: true)
                ]
                input[8] = Channel.of([ [:], [] ])
                input[9] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_nuc_stats.tsv", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("all files - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_trfinder.tsv", checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_aggregated_classification.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_genes.tsv", checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_taxonomy.tsv", checkIfExists: true)
                ]
                input[5] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_extra_hmm_hits.tsv", checkIfExists: true)
                ]
                input[6] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_completeness.tsv", checkIfExists: true)
                ]
                input[7] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_contamination.tsv", checkIfExists: true)
                ]
                input[8] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_tantan.tsv", checkIfExists: true)
                ]
                input[9] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/combinedata/test1_nuc_stats.tsv", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
