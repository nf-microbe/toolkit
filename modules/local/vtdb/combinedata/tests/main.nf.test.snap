{
    "no trfinder": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,64dc9002081d78f10c6485c77f76e504"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,64dc9002081d78f10c6485c77f76e504"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T11:49:20.697391232"
    },
    "no tantan": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,f0690f397295125f35a6937e038da8e7"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,f0690f397295125f35a6937e038da8e7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T11:50:44.357384516"
    },
    "no nucleotide stats": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,cbd6a874e861b42183bcd31d6a857d2f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,cbd6a874e861b42183bcd31d6a857d2f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T11:50:28.501406047"
    },
    "all files - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-15T07:24:23.673466483"
    },
    "no checkv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,35226081ac2cc45c7c916c78a254e01d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,35226081ac2cc45c7c916c78a254e01d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T11:50:12.748339278"
    },
    "no pyhmmer": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,9020603d59cc916cb2a719f2135cc548"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,9020603d59cc916cb2a719f2135cc548"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-15T07:26:50.461941683"
    },
    "no genomad": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,bd61aca0aa1a0a3bb3c3e35e88e3cca6"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,bd61aca0aa1a0a3bb3c3e35e88e3cca6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T11:49:38.406595851"
    },
    "all files": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,84001efed6d109561d86cbf06ba33251"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_combined_data.tsv:md5,84001efed6d109561d86cbf06ba33251"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2604bef49a2c80dc6dbd0527fd6e246"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T11:48:59.622529241"
    }
}